libs/gltfio/src/ResourceLoader.cpp

```diff
@@ -1061,4 +1011,45 @@ void ResourceLoader::updateBoundingBoxes(FFilamentAsset* asset) const {
     asset->mBoundingBox = assetBounds;
 }
 
+ResourceLoader* ResourceLoader::create(const ResourceConfiguration& config) {
+    return new FResourceLoader(config);
+}
+
+void ResourceLoader::destroy(ResourceLoader** loader) {
+    delete *loader;
+    *loader = nullptr;
+}
+
+void ResourceLoader::addResourceData(const char* uri, BufferDescriptor&& buffer) {
+    upcast(this)->addResourceData(uri, std::move(buffer));
```

Comment:
There's gotta be a better way of passing BufferDescriptor&& buffer through than using std::move(buffer) twice along the way.

Reply:
`std::move(buffer)` is essentially a no-op -- that's the correct way to do this.
`BufferDescriptor` can never be copied (because it holds a callback).

Reply:
Thanks for the reply.